### ISSUE TITLE
7.0.0 errors with EPIPE

### ISSUE THREAD

**Issue by danez** [imagemin/imagemin-mozjpeg#28]
We receive this error after updating to 7.0.0 on our ci server (Ubuntu 16)

```
Warning: Error: write EPIPE in file x.jpg
```

After downgrading to 6.0.0 it works again. It also works with newer version on macOS

Gruntfile.js
```js
...
             imagemin: {
                options: {
                    optimizationLevel: 7,
                    progressive: true,
                    interlaced: true,
                    svgoPlugins: [
                        { removeTitle: true },
                        { removeDesc: { removeAny: true } },
                    ],
                    use: [
                        imageminMozjpeg({ quality: 80 }),
                        imageminPngcrush({ reduce: false }),
                    ],
                },
                static: {
                    files: [
                        {
                            expand: true,
                            src: [
                                'webroot/images/**/*.{png,jpg,gif,svg}',
                            ],
                        },
                    ],
                },
...
```

installed packages:

```
├─ grunt-contrib-imagemin@2.0.1
├─ imagemin-gifsicle@5.2.0
├─ imagemin-jpegtran@5.0.2
├─ imagemin-mozjpeg@7.0.0
├─ imagemin-optipng@5.2.1
├─ imagemin-pngcrush@5.1.0
├─ imagemin-svgo@5.2.4
├─ imagemin@5.3.1
├─ mozjpeg@5.0.0
```

I guess it might be because of #16 which made it into 7.0.0 although it was reverted in #17, but the revert seems it was force pushed away or somehow else removed from history. Was that on purpose?

This is the image:

![3sat](https://user-images.githubusercontent.com/231804/35095735-e0016f5c-fc49-11e7-8d47-426c1b778278.jpg)

**Comment by magmel48** [comment]
have the same issue

**Comment by soptimizer** [comment]
sama issue

**Comment by adenvt** [comment]
same issue

**Comment by adenvt** [comment]
i not very sure, but fix after `sudo apt-get install libpng16-dev`

**Comment by kirylpl** [comment]
@adenvt Dziakui, advent.

**Comment by flexdinesh** [comment]
For anyone who's trying to get this to work in **Travis** build, 

`sudo apt-get install libpng16-dev` won't work as travis containers have a whitelist of packages - [ubuntu-trusty whitelist](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-trusty)

`libpng16-dev` is not in the whitelist yet and I have created a new whitelist request for `libpng16-dev` - [libpng16-dev whitelist request](https://github.com/travis-ci/travis-ci/issues/9515). So until it is whitelisted we won't be able to get it work work in trusty containers.

As an alternative, I switched to osx environment temporarily to get this to work. Add this line to you `travis.yml` file

`os: osx`

And everything will work like a charm! 🎉

**Comment by lekkerduidelijk** [comment]
I have the same issues, when running this from a GitLab CI runner. My buildchain uses the official Docker node ([node:6](https://github.com/nodejs/docker-node/blob/7fea7b033c5a292dd74199d2ba0fbb161f75aa0d/6/jessie/Dockerfile) and [node:8](https://github.com/nodejs/docker-node/blob/7fea7b033c5a292dd74199d2ba0fbb161f75aa0d/8/jessie/Dockerfile)) and both give the `Error: write EPIPE`.

**Tried**
[Installing libpng16-dev](https://github.com/imagemin/imagemin-mozjpeg/issues/28#issuecomment-382285717) with `sudo apt-get install libpng16-dev` doesn't work, since the Node Docker containers are based on Debian.

Also [installing `nasm`](https://github.com/tcoopman/image-webpack-loader/issues/142#issuecomment-389324023) doesn't work.

**Workaround**
When I exclude the `plugins: [imageminMozjpeg({ quality: 75 })]`, it works. 
When I downgrade to `"imagemin-mozjpeg": "~6.0.0"`, it works.

Note: I'm running this in a [Sage project](https://github.com/roots/sage/blob/master/package.json).

Possibly related: #26

**Comment by BenzLeung** [comment]
I have the same issue but I just found a new solution.

https://github.com/imagemin/imagemin-mozjpeg/blob/005b789088921493ea9968570562f59e10c5e838/package.json#L46

Just upgrade `mozjpeg` to the latest `6.0.0` version.

If you use `yarn` instead of `npm` , you can add the following statement to your project's package.json to temporarily resolve this issue:

```json
"resolutions": {
    "imagemin-mozjpeg/mozjpeg": "^6.0.0"
}
```

I hope the author will update as soon as possible.

**Comment by usamamashkoor** [comment]
same issue.

**Comment by yesworld** [comment]
have the same issue :(

**Comment by john-codeworx** [comment]
same issue, downgrading to ^6.0.0 fixed it.

**Comment by hua1995116** [comment]
I use ```npm i ``` it break when download imagemin-mozjpeg  .
```
/node_modules/mozjpeg/vendor/cjpeg` binary doesn't seem to work correctly
  ⚠ mozjpeg pre-build test failed
  ℹ compiling from source
```
But i use ```npm i  imagemin-mozjpeg@6.0.0``` it fixed.

**Comment by gbiryukov** [comment]
looks like this issue is solved in `8.0.0`

**Comment by kernel72** [comment]
Nope. Experiencing the same issue for some jpg files in `8.0.0`. After downgrade to `6.0.0` issue has gone.

**Comment by quinncomendant** [comment]
I have the same issue, and fixed it by downgrading to `6.0.0`. I tried to install `libpng16-dev`, but it is not available on CentOS 7.

> ```
> /node_modules/mozjpeg/vendor/cjpeg` binary doesn't seem to work correctly
>   ⚠ mozjpeg pre-build test failed
>   ℹ compiling from source
> ```

@hua1995116, I had this error, it was solved by `yum install libpng12-devel`.

**Comment by Sdcrow** [comment]
For those using the gulp version `gulp-imagemin` I faced this issue running `"gulp-imagemin": "2.4.0"` and I resolved it with the following steps

> npm cache clean -f
> npm install

I hope this helps someone
